### DISCUSSION TITLE
New blazor project taking into account blazor united

### DISCUSSION THREAD

**Discussion by FM1973** [dotnet/aspnetcore#50699] [Q&A] [answered]
Hello.

We want to start a new blazor project now (using .net 7) and take blazor united into account (we can´t use the .net 8 preview version).
What type of project would you suggest? Server or WASM. Which one of these would/will be easier to migrate to blazor united later on?

Thanks for your suggestions!

**Comment by marinasundstrom** [comment, marked as answer]
The new functionality in the "Blazor Web App" template is basically the evolution of what is in Blazor Server hosting mode - but with support for SSR, render modes, enhanced navigation etc. The rest of the Blazor features are available to all hosting model, incl. Blazor WASM.

So what should you start with? It depends on where you put most of your logic.

If you consider having a lot of logic on the client, start with Blazor WASM, and then you can easily attach the project to the server hosting project.

Otherwise, choose Server.

The minimal thing you will need to do, apart from learning how to deal with SSR (the harder part), is to change some things in your ``Program.cs``.

But if you are not planning to release very soon, why not already start with .NET 8 RC? 
It is already quite stable, and the final release is about 1-2 months anyway.
